scipy/interpolate/tests/test_bsplines.py

```diff
@@ -327,58 +325,61 @@ def test_derivative_method(self):
         xx = np.linspace(t[k], t[-k-1], 20)
         for j in range(1, k):
             b = b.derivative()
-            assert_allclose(b0(xx, j), b(xx), atol=1e-12, rtol=1e-12)
+            xp_assert_close(b0(xx, j), b(xx), atol=1e-12, rtol=1e-12)
 
     def test_antiderivative_method(self):
         b = _make_random_spline()
         t, c, k = b.tck
         xx = np.linspace(t[k], t[-k-1], 20)
-        assert_allclose(b.antiderivative().derivative()(xx),
+        xp_assert_close(b.antiderivative().derivative()(xx),
                         b(xx), atol=1e-14, rtol=1e-14)
 
         # repeat with N-D array for c
         c = np.c_[c, c, c]
         c = np.dstack((c, c))
         b = BSpline(t, c, k)
-        assert_allclose(b.antiderivative().derivative()(xx),
+        xp_assert_close(b.antiderivative().derivative()(xx),
                         b(xx), atol=1e-14, rtol=1e-14)
 
     def test_integral(self):
         b = BSpline.basis_element([0, 1, 2])  # x for x < 1 else 2 - x
-        assert_allclose(b.integrate(0, 1), 0.5)
-        assert_allclose(b.integrate(1, 0), -1 * 0.5)
-        assert_allclose(b.integrate(1, 0), -0.5)
+        xp_assert_close(b.integrate(0, 1), 0.5, check_0d=False)
```

Comment:
@ev-br I think this is an example of what @mdhaber was referring to in https://github.com/scipy/scipy/pull/21413#issuecomment-2329257608.

When this gets array API standard support, this line will fail the namespace check, because `0.5` will not have the same namespace as (say) array-api-strict. So at that stage, we will want to make this `xp.asarray(0.5)`. I think having to add `np.asarray` is not so bad here given that it is only 1 character away from what we will end up with eventually?

Reply:
Sure. Is this the thing which keeps this PR from landing?

Reply:
Sorry, I haven't taken a look through the diff yet!

Reply:
OK, done.